src/shogun/classifier/LDA.cpp

```diff
@@ -91,67 +94,81 @@ bool CLDA::train_machine(CFeatures *data)
 
 template <typename ST>
 bool CLDA::train_machine_templated(SGVector<int32_t> train_labels, CFeatures *data)
-{	
-	SGMatrix<ST>feature_matrix=((CDenseFeatures<ST>*)features)
-										->get_feature_matrix();
-	int32_t num_feat=feature_matrix.num_rows;
-	int32_t num_vec=feature_matrix.num_cols;
-
-	SGVector<int32_t> classidx_neg(num_vec);
-	SGVector<int32_t> classidx_pos(num_vec);
-
-	int32_t i=0;
-	int32_t num_neg=0;
-	int32_t num_pos=0;
+{
+	SGMatrix<ST> feature_matrix =
+	    ((CDenseFeatures<ST>*)features)->get_feature_matrix();
+	index_t num_feat = feature_matrix.num_rows;
+	index_t num_vec = feature_matrix.num_cols;
 
-	for(i=0; i<train_labels.vlen; i++)
+	if (m_method == SVD_LDA || (m_method == AUTO_LDA && num_vec <= num_feat))
```

Comment:
You could make this more readable via pulling the `m_method == AUTO_LDA && num_vec <= num_feat` out into a variable and call it `lda_more_efficient`